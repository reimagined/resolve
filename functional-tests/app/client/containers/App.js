import React from 'react'
import { Helmet } from 'react-helmet'

const App = ({ staticPath }) => {
  const stylesheetLink = {
    rel: 'stylesheet',
    type: 'text/css',
    href: `${staticPath}/bootstrap.min.css`,
  }
  const faviconLink = {
    rel: 'icon',
    type: 'image/png',
    href: `${staticPath}/favicon.ico`,
  }
  const links = [stylesheetLink, faviconLink]
  const meta = {
    name: 'viewport',
    content: 'width=device-width, initial-scale=1',
  }

  return (
    <div>
      <Helmet title="Functional Tests App" link={links} meta={[meta]} />
      <h2 align="center">Basic tests</h2>
      <h2 align="center">View model tests</h2>
    </div>
  )
}

export default App
