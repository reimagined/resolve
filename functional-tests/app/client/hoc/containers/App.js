import React from 'react'
import { Helmet } from 'react-helmet'

const App = ({ staticPath }) => {
  const stylesheetLink = {
    rel: 'stylesheet',
    type: 'text/css',
    href: `${staticPath}/bootstrap.min.css`,
  }
  const faviconLink = {
    rel: 'icon',
    type: 'image/png',
    href: `${staticPath}/favicon.ico`,
  }
  const links = [stylesheetLink, faviconLink]
  const meta = {
    name: 'viewport',
    content: 'width=device-width, initial-scale=1',
  }

  return (
    <div>
      <Helmet title="HOC client" link={links} meta={[meta]} />
      <h2 align="center">HOC client</h2>
    </div>
  )
}

export default App
