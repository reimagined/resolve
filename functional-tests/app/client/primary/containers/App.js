import React from 'react'
import { Helmet } from 'react-helmet'
import { useStaticResolver } from 'resolve-react-hooks'

const App = ({ version, children }) => {
  const resolveStatic = useStaticResolver()

  const stylesheetLink = {
    rel: 'stylesheet',
    type: 'text/css',
    href: resolveStatic(`/bootstrap.min.css`),
  }
  const faviconLink = {
    rel: 'icon',
    type: 'image/png',
    href: resolveStatic(`/favicon.ico`),
  }
  const links = [stylesheetLink, faviconLink]
  const meta = {
    name: 'viewport',
    content: 'width=device-width, initial-scale=1',
  }

  return (
    <div>
      <Helmet title="React Hooks Tests" link={links} meta={[meta]} />
      <h6 align="left">
        {version != null && version.length > 0 ? `Version ${version}` : ``}
      </h6>
      {children}
    </div>
  )
}

export { App }
