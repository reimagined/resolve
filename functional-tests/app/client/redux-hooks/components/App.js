import React from 'react'

import Header from './Header'

const App = ({ children }) => (
  <div>
    <Header
      title="HOC tests"
      favicon="/favicon.ico"
      css={['/bootstrap.min.css', '/fontawesome.min.css', '/style.css']}
    />
    {children}
  </div>
)

export default App
