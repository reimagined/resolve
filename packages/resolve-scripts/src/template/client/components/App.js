import React from 'react'

const App = () => (
  <div>
    <h1>Hello, world!</h1>
  </div>
)

export default App
