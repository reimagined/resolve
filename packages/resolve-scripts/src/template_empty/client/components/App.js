import React from 'react';

const RootComponent = () =>
    <div>
        <h1>Hello, world!</h1>
    </div>;

export default RootComponent;
