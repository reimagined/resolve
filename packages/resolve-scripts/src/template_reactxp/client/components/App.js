import React from 'react'
import {
  Component,
  Animated,
  Styles,
  View,
  Text,
  ScrollView,
  Button,
  TextInput
} from 'reactxp'
import { bindActionCreators } from 'redux'
import { connect } from 'resolve-redux'
import ToggleSwitch from './ToggleSwitch'

import actions from '../actions'

const viewModelName = 'Todos'
const aggregateId = 'root-id'

const styles = {
  scroll: Styles.createScrollViewStyle({
    alignSelf: 'stretch',
    backgroundColor: '#f5fcff'
  }),
  container: Styles.createViewStyle({
    padding: 16,
    justifyContent: 'center',
    alignItems: 'center'
  }),
  header: Styles.createTextStyle({
    fontSize: 32,
    marginBottom: 12
  }),
  listRow: Styles.createViewStyle({
    flexDirection: 'row',
    alignItems: 'center'
  }),
  todoIndex: Styles.createViewStyle({
    marginRight: 10
  }),
  todoText: Styles.createViewStyle({
    marginLeft: 20
  }),
  roundButton: Styles.createViewStyle({
    margin: 16,
    borderRadius: 16,
    backgroundColor: '#7d88a9'
  }),
  removeButton: Styles.createViewStyle({
    marginLeft: 5,
    backgroundColor: '#ddd'
  }),
  buttonText: Styles.createViewStyle({
    fontSize: 16,
    marginVertical: 6,
    marginHorizontal: 12,
    color: 'white'
  }),
  newTodoContainer: Styles.createViewStyle({
    flexDirection: 'row',
    alignItems: 'center'
  }),
  textInput: Styles.createTextInputStyle({
    borderWidth: 1,
    borderColor: 'gray',
    width: 100
  })
}

class App extends Component {
  constructor(props) {
    super(props)
    this._translationValue = Animated.createValue(-100)
    this._animatedStyle = Styles.createAnimatedTextStyle({
      transform: [
        {
          translateY: this._translationValue
        }
      ]
    })
    this.state = { newTodo: '' }
  }

  componentDidMount() {
    let animation = Animated.timing(this._translationValue, {
      toValue: 0,
      easing: Animated.Easing.OutBack(),
      duration: 500
    })

    animation.start()
  }

  render() {
    const {
      todos,
      createItem,
      toggleItem,
      removeItem,
      aggregateId
    } = this.props

    return (
      <ScrollView style={styles.scroll}>
        <View style={styles.container}>
          <Text style={styles.header}>TODO</Text>
          <View style={styles.list}>
            {Object.keys(todos).map((id, index) => (
              <View key={id} style={styles.listRow}>
                <Text style={styles.todoIndex}>{`${index + 1}.`}</Text>
                <ToggleSwitch
                  value={todos[id].checked}
                  onChange={toggleItem.bind(null, aggregateId, { id })}
                />
                <Text style={styles.todoText}>{todos[id].text}</Text>
                <Button
                  style={styles.removeButton}
                  onPress={removeItem.bind(null, aggregateId, { id })}
                >
                  <Text>{' X '}</Text>
                </Button>
              </View>
            ))}
          </View>

          <View style={styles.newTodoContainer}>
            <Button
              style={styles.roundButton}
              onPress={() => {
                this.setState({ newTodo: '' })
                createItem(aggregateId, {
                  text: this.state.newTodo,
                  id: Date.now()
                })
              }}
            >
              <Text style={styles.buttonText}>Add Todo</Text>
            </Button>
            <TextInput
              style={styles.textInput}
              value={this.state.newTodo}
              onChangeText={newValue => this.setState({ newTodo: newValue })}
            />
          </View>
        </View>
      </ScrollView>
    )
  }
}

const mapStateToProps = state => ({
  viewModelName,
  aggregateId,
  todos: state[viewModelName][aggregateId]
})

const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(App)
