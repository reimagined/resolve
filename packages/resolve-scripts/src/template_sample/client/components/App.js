import React from 'react'
import styled from 'styled-components'
import Filters from './Filters'
import AddTodo from '../containers/AddTodo'
import VisibleTodoList from '../containers/VisibleTodoList'

const PageContent = styled.div`
  background-color: #f1f1f1;
  width: 300px;
  margin: 0 auto;
  font-size: 10pt;
  font-family: Verdana, Geneva, sans-serif;
`

const PageHeader = styled.div`
  color: #fff;
  background-color: #3949ab;
  display: flex;
  align-items: center;
  padding: 8px;
`

const PageTitle = styled.div`
  text-align: center;
  font-weight: bold;
  flex-basis: 100%;
`

const PageFooter = styled.div`
  background-color: #ddd;
  text-align: center;
  padding: 6px;
`

const FooterLink = styled.a`
  color: #777;
  text-decoration: underline;
`

const App = () => (
  <PageContent>
    <PageHeader>
      <img src="/static/reSolve-logo.svg" width="18" height="18" alt="" />
      <PageTitle>Resolve TodoList</PageTitle>
    </PageHeader>
    <AddTodo />
    <VisibleTodoList />
    <Filters />
    <PageFooter>
      <FooterLink href="https://github.com/reimagined/resolve" target="_blank">
        reimagined/resolve
      </FooterLink>
    </PageFooter>
  </PageContent>
)

export default App
