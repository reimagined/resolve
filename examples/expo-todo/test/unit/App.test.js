import React from 'react'
import renderer from 'react-test-renderer'

import { App } from '../../client/containers/App'

test('renders correctly', () => {
  const tree = renderer
    .create(
      <App
        todos={{
          id1: {
            text: 'text1',
            checked: true
          },
          id2: {
            text: 'text2',
            checked: false
          }
        }}
        createItem={() => {}}
        toggleItem={() => {}}
        removeItem={() => {}}
        aggregateId={() => {}}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
