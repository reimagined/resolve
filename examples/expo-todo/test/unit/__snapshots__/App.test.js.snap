// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-default"
    >
      <div
        className="container"
      >
        <p
          className="navbar-text"
        >
          <img
            className=""
            src="/resolve-logo.png"
          />
           Todo Example
        </p>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
        >
          <p
            className="navbar-text navbar-right"
          >
            <a
              className="navbar-link"
              href="https://facebook.com/resolvejs/"
            >
              <img
                className=""
                src="/fb-logo.png"
              />
            </a>
          </p>
          <p
            className="navbar-text navbar-right"
          >
            <a
              className="navbar-link"
              href="https://twitter.com/resolvejs"
            >
              <img
                className=""
                src="/twitter-logo.png"
              />
            </a>
          </p>
          <p
            className="navbar-text navbar-right"
          >
            <a
              className="navbar-link"
              href="https://github.com/reimagined/resolve"
            >
              <img
                className=""
                src="/github-logo.png"
              />
            </a>
          </p>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="example-wrapper"
  >
    <h1>
      Task's List
    </h1>
    <ul
      className="example-list list-group"
    >
      <li
        className="list-group-item"
      >
        <label
          className="checkbox-inline"
          style={undefined}
          title=""
        >
          <input
            checked={true}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          text1
        </label>
        <img
          className="example-close-button"
          onClick={[Function]}
          src="/close-button.png"
        />
      </li>
      <li
        className="list-group-item"
      >
        <label
          className="checkbox-inline"
          style={undefined}
          title=""
        >
          <input
            checked={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          text2
        </label>
        <img
          className="example-close-button"
          onClick={[Function]}
          src="/close-button.png"
        />
      </li>
    </ul>
    <form
      className="example-form form-inline"
    >
      <input
        className="example-form-control form-control"
        id={undefined}
        onKeyPress={[Function]}
        placeholder="New Task"
        type="text"
      />
      <button
        className="example-button btn btn-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Add Task
      </button>
    </form>
  </div>
</div>
`;
